src/Main.java

```diff
@@ -0,0 +1,264 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException {
+
+        try (Scanner scanner = new Scanner(System.in)) {
+
+            String continueOrGameOver;
+            String playerName;
+            int numberOfBoardsWon = 0;
+
+            Map<Character, String> playersMap = new HashMap<>();
+            Map<String, Integer> mapOfWinningBoards = new HashMap<>();
+
+            System.out.println("Please enter the name of the first player(as player X): ");
+            playerName = scanner.nextLine();
+            playersMap.put('X', playerName);
+            mapOfWinningBoards.put(playerName, numberOfBoardsWon);
+
+            System.out.println("Please enter the name of the second player(as player 0): ");
+            playerName = scanner.nextLine();
+            playersMap.put('0', playerName);
+            mapOfWinningBoards.put(playerName, numberOfBoardsWon);
+
+            do {
+                int boardDimension;
+                char activePlayer = 'X';
+                int i;
+                int j;
+
+                System.out.println("Please provide the board dimension (min 3, max 10): ");
+                boardDimension = scanner.nextInt();
```

Comment:
to można objąć w klauzule try catch bo ktoś mozę wpisać literę i wtedy program sie wyłoży